number_theory/numeric_unittest.cpp

```diff
@@ -0,0 +1,42 @@
+#include <stdint.h>
+
+#include <cmath>
+#include <limits>
+#include <random>
+
+#include <gtest/gtest.h>
+
+#include "number_theory/numeric.h"
+
+using namespace tql::number_theory;
+
+template <class T>
```

Comment:
Maybe we can consider putting this function into unnamed namespace to make it private to the file. Here is an example from chromium source code: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ash/arc/file_system_watcher/file_system_scanner_unittest.cc